Feature: Implement experiment run name update functionality

gustavoteixeirah opened the pull request:
Added functionality to update the name of an experiment run by modifying the methods in FutureExperimentRunServiceImpl and FutureExperimentRunDAO classes. The changes permit the updating of the experiment run name in the database. This functionality was implemented in response to the need for modification of experiment run details after their creation. Also, optimized the import statements in FutureExperimentRunDAO class for better readability and maintenance.

<!-- Example Title: "fix: [JIRA-123] Allow creation of groups with no members" -->
## Impact and Context

## Risks and Area of Effect
- [ ] Is this a breaking change?

## Testing
- [x] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

Commit:
Implement experiment run name update functionality

Added functionality to update the name of an experiment run by modifying the methods in FutureExperimentRunServiceImpl and FutureExperimentRunDAO classes. The changes permit the updating of the experiment run name in the database. This functionality was implemented in response to the need for modification of experiment run details after their creation. Also, optimized the import statements in FutureExperimentRunDAO class for better readability and maintenance.

jkwatson-verta commented on backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:
```diff
@@ -1876,4 +1789,28 @@ public void deleteLogVersionedInputs(Session session, List<Long> repoIds) {
+                jdbi.withHandle(
+                    handle -> {
+                      String updateQueryString =
+                          "UPDATE experiment_run SET name=:name, date_updated=:date_updated WHERE id=:runId";
```
maybe need to update the version, as well?  Not sure...

gustavoteixeirah replied:
I've added, I saw that other method are doing it. ✅

jkwatson-verta commented on backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:
```diff
@@ -1876,4 +1789,28 @@ public void deleteLogVersionedInputs(Session session, List<Long> repoIds) {
     LOGGER.debug("ExperimentRun versioning deleted successfully");
   }
+
+  public InternalFuture<Empty> updateExperimentRunName(UpdateExperimentRunName request) {
```
might as well make new code use `Future` while were changing things

gustavoteixeirah replied:
Adjusted ✅

Commit:
Update experiment run name functionality

Implemented new methods updateModifiedTimestampV2 and
updateVersionNumberV2 in FutureExperimentRunDAO for update operation of
experiment run name. Refactored updateExperimentRunName method to use
the new implementations. Optimal import statements also added as part of
readability improvement. Aim is to provide efficient way of updating
experiment run name after creation.

jkwatson-verta commented on backend/server/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:
```diff
@@ -1876,4 +1817,29 @@ public void deleteLogVersionedInputs(Session session, List<Long> repoIds) {
+                          .execute();
+                    }))
+        .thenCompose(unused -> updateModifiedTimestampV2(runId, now))
+        .thenCompose(unused -> updateVersionNumberV2(runId));
```
I wouldn't do this like this. Just put the updates inline into the query... Let's keep it simple unless we have some reason to think it doesn't need to be simple.

gustavoteixeirah replied:
Updated ✅

jkwatson-verta commented:
needs a unit test, and simplify the update into a single sql update, and this should be fine

Commit:
Add validation and tests for ExperimentRun's name update

Added validation in FutureExperimentRunServiceImpl to ensure that the
experiment run's name is not blank. An exception is thrown if a name
argument isn't present when updating the experiment run's name. To
ensure the proper implementation, added corresponding tests in
ExperimentRunTest to verify correct results and the absence of any
exceptions when the name is properly updated. Also tested the scenario
when the name argument is missing.

Commit:
Add versioning in ExperimentRun's name update

Incremented the version number each time the name of the experiment run
is updated. This enhancement was done to track the changes in the name
over time. Removed unneeded methods `updateModifiedTimestampV2` and
`updateVersionNumberV2` as the changes are handled within the `UPDATE`
sql statement. Verified these changes in the `ExperimentRunTest` by
asserting that the version number increases by one after a name update.

jkwatson-verta reviewed (approved):
:shipit:

Commit:
removed deprecated
